api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,313 @@
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/variant.h"
+#include "opentelemetry/trace/default_span.h"
+#include "opentelemetry/trace/key_value_iterable.h"
+#include "opentelemetry/trace/propagation/http_text_format.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/span_context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace
+{
+namespace propagation
+{
+static const nostd::string_view kTraceParent = "traceparent";
+static const nostd::string_view kTraceState  = "tracestate";
+static const int kVersionBytes               = 2;
+static const int kTraceIdBytes               = 32;
+static const int kSpanIdBytes                = 16;
+static const int kTraceFlagBytes             = 2;
+static const int kTraceDelimiterBytes        = 3;
+static const int kHeaderSize =
+    kVersionBytes + kTraceIdBytes + kSpanIdBytes + kTraceFlagBytes + kTraceDelimiterBytes;
+static const int kTraceStateMaxMembers    = 32;
+static const int kHeaderElementLengths[4] = {2, 32, 16, 2};
+
+// The HttpTraceContext provides methods to extract and inject
+// context into headers of HTTP requests with traces.
+// Example:
+//    HttpTraceContext.inject(setter,&carrier,&context);
+//    HttpTraceContext.extract(getter,&carrier,&context);
+template <typename T>
+class HttpTraceContext : public HTTPTextFormat<T>
+{
+public:
+  // Rules that manages how context will be extracted from carrier.
+  using Getter = nostd::string_view (*)(const T &carrier, nostd::string_view trace_type);
+
+  // Rules that manages how context will be injected to carrier.
+  using Setter = void (*)(T &carrier,
+                          nostd::string_view trace_type,
+                          nostd::string_view trace_description);
+
+  void Inject(Setter setter, T &carrier, const context::Context &context) noexcept override
+  {
+    SpanContext span_context = GetCurrentSpan(context)->GetContext();
+    if (!span_context.IsValid())
+    {
+      return;
+    }
+    InjectImpl(setter, carrier, span_context);
+  }
+
+  context::Context Extract(Getter getter,
+                           const T &carrier,
+                           context::Context &context) noexcept override
+  {
+    SpanContext span_context    = ExtractImpl(getter, carrier);
+    nostd::string_view span_key = "current-span";
+    nostd::shared_ptr<Span> sp{new DefaultSpan(span_context)};
+    return context.SetValue(span_key, sp);
+  }
+
+  static Span *GetCurrentSpan(const context::Context &context)
+  {
+    const nostd::string_view span_key = "current-span";
+    context::Context ctx(context);
+    nostd::shared_ptr<Span> span = nostd::get<nostd::shared_ptr<Span>>(ctx.GetValue(span_key));
+    return (span.get());
+  }
+
+  static TraceId GenerateTraceIdFromString(nostd::string_view trace_id)
+  {
+    const char *trc_id = trace_id.begin();
+    uint8_t buf[kTraceIdBytes / 2];
+    int tmp;
+    for (int i = 0; i < kTraceIdBytes; i++)
+    {
+      tmp = CharToInt(*trc_id);
```

Comment:
```suggestion
    for (int i = 0; i < kTraceIdBytes; i++)
    {
      int tmp = CharToInt(*trc_id);
```

Narrowing the scope of local variables makes them easier to keep track of, and initializing at declaration means it's never uninitialized.

Reply:
I resolved it on my own branch